### ISSUE TITLE
Support logging in multiple users (of the same team) when online

### ISSUE THREAD

**Issue by LZRS** [opensrp/fhircore#2330]
**Describe the feature request.**
Support for multiple members of the same team logging in on the same device without requiring data to be purged/fresh synced.

**Additional context**
Team implies the sync strategy assigned by the health provider at deployment e.g Sync By Team, CareTeam, Organisation, Location

Relates to https://github.com/opensrp/fhircore/issues/2283#issuecomment-1536182526

**Acceptance criteria**
- only allow user to change the username when online, and disable when offline with message https://github.com/opensrp/fhircore/issues/2283#issuecomment-1529926067
- only allow user with the same team (defined as same sync strategy) as currently logged-in, to log in
-  

**Implementation plan (For Engineers)**
Documented [here](https://docs.google.com/document/d/1ZfdKZMzh6kObrf0WxwGxTuiHzFEzYdu_WYoZEbfXQrI/edit?usp=sharing)

**Comment by LZRS** [comment]
@pld here's the issue that would track the work on multi-tenancy. On the question of unsynced data, I'm still not sure how to handle that, since we'd only allow users of the same team to login and they could probably share data

cc @ndegwamartin @f-odhiambo

**Comment by pld** [comment]
Agree, not an issue if we only allow users who have access to the same set of data to log-in then we don't have to worry about unsynced data.

If access to data is only controlled by the CareTeam, then verifying that the user trying to log in is a member of that care team would be sufficient.

If people in the same CareTeam can have varying data access, then that would not work.

**Comment by pld** [comment]
The ideas in the PR look good, I do want us all to think really hard about any edge cases or scenarios where people on the same team should not have access to the same data, think it over, please note any ideas or concerns you have here.

**Comment by ellykits** [comment]
There is a possibility for two users (belonging to the same team) to have different permissions/keycloak roles.  @dubdabasoduba will this be a concern?

**Comment by pld** [comment]
Yes that is a problem, they must have the same permission (data access rights) in this modelOn Oct 6, 2023, at 08:36, Elly Kitoto ***@***.***> wrote:﻿
There is a possibility for two users (belonging to the same team) to have different permissions/keycloak roles.  @dubdabasoduba will this be a concern?

**Comment by LZRS** [comment]
> > Yes that is a problem, they must have the same permission (data access rights) in this modelOn Oct 6, 2023, at 08:36, Elly Kitoto _**@**_._**> wrote:﻿ There is a possibility for two users (belonging to the same team) to have different permissions/keycloak roles. @dubdabasoduba will this be a concern? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: _**@**_.**_>
> 
> @LZRS Is it possible to add this check for multi-user login implementation?

Yeah I think it should be possible...we'd probably need to check and compare out roles as encoded in the jwt tokens
